classes/announcement/Collector.inc.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * @file classes/announcement/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class announcement
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\announcement;
+
+use HookRegistry;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+
+class Collector implements CollectorInterface
+{
+    public DAO $dao;
+    public array $contextIds = [];
```

Comment:
I think type declarations are only available for class properties as of PHP7.4. Are you relying on this, or is it something we can defer a little bit? (I'm not 100% averse to bumping our baseline to PHP7.4, but I'd rather not if it's not giving us much in return.)

Reply:
:sob: You're right! They're not totally vital, it's just nice to not have to do `/** @var  ...*/`. it looks like PHP 7.3 is EOL in December, which may be just a bit too late for 3.4. I'll remove them.